omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -64,13 +46,29 @@ export default class LatestObsForConceptListController {
             '8d4a591e-c2cc-11de-8d13-0010c6dffd0f',
             '8d4a5af4-c2cc-11de-8d13-0010c6dffd0f'].indexOf(obs.concept.datatype.uuid) > -1) {
             //If value is date, time or datetime
-            var date = this.$filter('date')(new Date(obs.value), this.config.dateFormat);
-            return date;
+            return this.$filter('date')(new Date(obs.value), this.config.dateFormat);
         } else if (angular.isDefined(obs.value.display)) {
             //If value is a concept
-             return obs.value.display;
+             return this.getConceptName(obs.value, this.config.conceptNameType);
         } else {
             return obs.value;
         }
     }
+
+    getConceptName(concept, nameType) {
+        const names = concept.names.filter(n => !n.voided && n.locale === this.config.locale);
+        const fsn = names.filter(n => n.conceptNameType === "FULLY_SPECIFIED")[0];
+        const short = names.filter(n => n.conceptNameType === "SHORT")[0];
+        const shortEn = concept.names.filter(n => !n.voided && n.locale === 'en' && n.conceptNameType === "SHORT")[0];
```

Comment:
haven't we already filtered out locales that don't match the current locale at this point?

Reply:
That's the list in `names`, which `fsn`, `short`, and `preferred` draw from. `shortEn` is pulled from the original list, which contains all names in all locales.

Reply:
Oh, right, I thought I double-checked that, but I guess not... :)